ros-build-essential/ros-build-essential.spec

```diff
@@ -0,0 +1,25 @@
+Name: ros-build-essential
+Version: 1.0.0
+Summary: ROS Infrastructure variant including packages that are expected to be present when building any ROS packages.
+Release: %{?dist}%{?release_suffix}
+BuildArch: noarch
+License: Apache-2.0
+URL: https://github.com/ros-infrastructure/infra-variants
+
+# TODO add packages from mock's default configuration.
+Requires: cmake, g++, git, make, python3
+
+%description
+ROS Infrastructure variant including packages that are expected to be present when building any ROS packages.
+
+%install
+install -p -D -m 644 README.md %{buildroot}/%{doc}/%{name}/README.md
+install -p -D -m 644 copyright %{buildroot}/%{doc}/%{name}/LICENSE
```

Comment:
@cottsay this is where I'm currently stuck when using rpmbuild as I don't know how to get these files which are siblings of the spec file on disk into the rpmbuild context.

Reply:
Here's what you'll want:
1. List the sources in the header, usually after the `URL` tag:
   ```
   Source0: README.md
   Source1: LICENSE
   ```
2. Drop the explicit `install` invocations in the `%install` section
3. Update the `%doc` and `%license` macros in the `%files` section to pull directly from the `%{SOURCE}` entries:
   ```
   %files
   %doc %{SOURCE0}
   %license %{SOURCE1}
   ```

Reply:
Thanks for the guidance. I implemented it in [e190a77](https://github.com/ros-infrastructure/infra-variants/pull/1/commits/e190a776ec2885106112050b2f0d14ec6cad0dec) attributing you as co-author. I'm still getting errors during the build, I've re-enabled RPM builds in the GitHub Actions workflow so you can see the error I'm getting there.

Reply:
Alright, I was wrong about some of this. This should get the spec file working. I'll make another comment about how to actually get it to build.

It seems that `%doc` and `%license` can only pull files from the `BUILD` directory, not from `SOURCES`. I recommend adding the following `%prep` section and changing the `%files` to look like this:
```
%prep
cp -a %{SOURCE0} %{SOURCE1} .
```

```
%files
%doc README.md
%license copyright
```

Reply:
Thanks, with this change and the proper configuration of _topdir and _sourcedir this appears to do the trick!